hostinfo/nginx_config.lua

```diff
@@ -203,10 +202,20 @@ function Info:_run(callback)
             elseif type(listen) == 'table' then
               table.insert(listen, data.listen)
             end
-          elseif domain then
+          else
             listen = data.listen
           end
-          vhost[domain]['listen'] = listen
+        end
+
+        if domain then
+          vhost[domain] = {}
+          vhost[domain]['domain'] = domain
+          if listen then
+            vhost[domain]['listen'] = listen
```

Comment:
we should probably do: vhost[domain]['listen'] = listen or '<UNDEFINED>'

ame with docroot